### PR TITLE
Task 54-57: BE: Implement CRUD API for product comment

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/comment/comment.controller.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    return this.commentService.getComments(productId);
+  }
+
+  @UseGuards(AuthGuard('jwt'))
+  @ApiBearerAuth('JWT-auth')
+  @Put(':id')
+  update(
+    @Param('id') id: string,
+    @Request() req,
+    @Body() UpdateCommentDto: UpdateCommentDto,
```
Comment (VYaniv) [inline src/comment/comment.controller.ts]: `UpdateCommentDto: UpdateCommentDto` -> `updateCommentDto: UpdateCommentDto`

File: src/comment/comment.service.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const comment = await this.commentModel.findOne({ _id: commentId, userId });
+
+    if (!comment) {
+      throw new NotFoundException('Comment not found or unauthorized');
+    }
+
+    comment.text = updateCommentDto.text;
+    comment.isUpdated = true;
+
+    return comment.save();
```
Comment (VYaniv) [inline src/comment/comment.service.ts]: We also need to compare the user ids, to be sure that only owner can update it

File: src/comment/comment.service.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  ): Promise<{ message: string }> {
+    const result = await this.commentModel.deleteOne({
+      _id: commentId,
+      userId,
+    });
+
+    if (result.deletedCount === 0) {
+      throw new NotFoundException('Comment not found or unauthorized');
+    }
+
```
Comment (VYaniv) [inline src/comment/comment.service.ts]: In my opinion for removing we need it as well

File: src/comment/comment.controller.ts (hunk 1 above)
Comment (di-kinosian) [reply src/comment/comment.controller.ts]: done

File: src/comment/comment.service.ts (hunk 2 above)
Comment (di-kinosian) [reply src/comment/comment.service.ts]: done

File: src/comment/comment.service.ts (hunk 3 above)
Comment (di-kinosian) [reply src/comment/comment.service.ts]: done

**Review by VYaniv — APPROVED** [review]

**Pull request merged**